cmd/cluster-capi-operator/main.go

```diff
@@ -243,79 +268,88 @@ func main() {
 		} else {
-			setupReconcilers(mgr, infra, platform, &azurev1.AzureCluster{}, containerImages, applyClient, apiextensionsClient, managedNamespace)
+			// The ClusterOperator Controller must run in all cases.
+			setupReconcilers(mgr, infra, platform, &azurev1.AzureCluster{}, containerImages, applyClient, apiextensionsClient, currentFeatureGates, managedNamespace, sourceNamespace)
 			setupWebhooks(mgr)
 		}
 	case configv1.PowerVSPlatformType:
-		setupReconcilers(mgr, infra, platform, &ibmpowervsv1.IBMPowerVSCluster{}, containerImages, applyClient, apiextensionsClient, managedNamespace)
+		setupReconcilers(mgr, infra, platform, &ibmpowervsv1.IBMPowerVSCluster{}, containerImages, applyClient, apiextensionsClient, currentFeatureGates, managedNamespace, sourceNamespace)
 		setupWebhooks(mgr)
 	case configv1.VSpherePlatformType:
-		setupReconcilers(mgr, infra, platform, &vspherev1.VSphereCluster{}, containerImages, applyClient, apiextensionsClient, managedNamespace)
+		setupReconcilers(mgr, infra, platform, &vspherev1.VSphereCluster{}, containerImages, applyClient, apiextensionsClient, currentFeatureGates, managedNamespace, sourceNamespace)
 		setupWebhooks(mgr)
 	case configv1.OpenStackPlatformType:
-		setupReconcilers(mgr, infra, platform, &openstackv1.OpenStackCluster{}, containerImages, applyClient, apiextensionsClient, managedNamespace)
+		setupReconcilers(mgr, infra, platform, &openstackv1.OpenStackCluster{}, containerImages, applyClient, apiextensionsClient, currentFeatureGates, managedNamespace, sourceNamespace)
 		setupWebhooks(mgr)
 	default:
 		klog.Infof("Detected platform %q is not supported, skipping capi controllers setup", platform)
-		setupUnsupportedController(mgr, managedNamespace)
+		// The ClusterOperator Controller must run under all circumstances as it manages the ClusterOperator object for this operator.
+		setupClusterOperatorController(mgr, managedNamespace, currentFeatureGates, true)
 	}
 }
 
-func setupReconcilers(mgr manager.Manager, infra *configv1.Infrastructure, platform configv1.PlatformType, infraClusterObject client.Object, containerImages map[string]string, applyClient *kubernetes.Clientset, apiextensionsClient *apiextensionsclient.Clientset, managedNamespace string) {
-	if err := (&corecluster.CoreClusterReconciler{
-		ClusterOperatorStatusClient: getClusterOperatorStatusClient(mgr, "cluster-capi-operator-cluster-resource-controller", managedNamespace),
+func setupReconcilers(mgr manager.Manager, infra *configv1.Infrastructure, platform configv1.PlatformType, infraClusterObject client.Object, containerImages map[string]string,
+	applyClient *kubernetes.Clientset, apiextensionsClient *apiextensionsclient.Clientset, currentFeatureGates featuregates.FeatureGate, managedNamespace, sourceNamespace string) {
+	// The ClusterOperator Controller must run under all circumstances as it manages the ClusterOperator object for this operator.
+	setupClusterOperatorController(mgr, managedNamespace, currentFeatureGates, false)
```

Comment:
Yeah, as with previous comment, maybe move this up a level and before the switch to avoid calling it in three different places if it must always run